src/grasshopper/lib/util/shapes.py

```diff
@@ -196,3 +197,52 @@ def __init__(self, *args, **kwargs):
             },
         ]
         super().__init__(*args, **kwargs)
+
+class Customstages(Default):  # noqa E501
+    """
+    Stolen from this set of examples as part of the locust.io documentation.
+    https://github.com/locustio/locust/blob/master/examples/custom_shape/stages.py
+
+    Keyword arguments:
+        stages -- A list of dicts, each representing a stage with the following keys:
```

Comment:
this class takes a keyword argument `stages` that is a json string with this structure. it's important to note that.

Reply:
Updated